narwhal/primary/src/synchronizer.rs

```diff
@@ -381,14 +461,22 @@ impl Synchronizer {
         certificate: Certificate,
         network: &Network,
         sanitize: bool,
+        early_suspend: bool,
     ) -> DagResult<()> {
         let digest = certificate.digest();
         if self.inner.certificate_store.contains(&digest)? {
             trace!("Certificate {digest:?} has already been processed. Skip processing.");
             self.inner.metrics.duplicate_certificates_processed.inc();
             return Ok(());
         }
-
+        // Ensure parents are checked if !early_suspend.
+        // See comments above `try_accept_fetched_certificate()` for details.
+        if early_suspend {
+            if let Some(notify) = self.inner.state.lock().await.check_suspended(&digest) {
+                trace!("Certificate {digest:?} is still suspended. Skip processing.");
```

Comment:
Maybe we could add a metric here as well? Before we used the `certificates_suspended` for every certificate we tried process (irrespective where it came from) and failed the parent checks - now might not reach that point and I would be curious to see how frequently we hit those cases

Reply:
Sure, incrementing `certificates_suspended` with different `reason` here.